llvm/include/llvm/SandboxIR/Type.h

```diff
@@ -0,0 +1,288 @@
+  /// includes struct and array types, but does not include vector types.
+  bool isAggregateType() const { return LLVMTy->isAggregateType(); }
+
+  /// Return true if it makes sense to take the size of this type. To get the
+  /// actual size for a particular target, it is reasonable to use the
+  /// DataLayout subsystem to do this.
+  bool isSized(SmallPtrSetImpl<Type *> *Visited = nullptr) const {
+    SmallPtrSet<llvm::Type *, 8> LLVMVisited;
+    LLVMVisited.reserve(Visited->size());
+    for (Type *Ty : *Visited)
+      LLVMVisited.insert(Ty->LLVMTy);
+    return LLVMTy->isSized(&LLVMVisited);
+  }
+
+  /// Return the basic size of this type if it is a primitive type. These are
+  /// fixed by LLVM and are not target-dependent.
+  /// This will return zero if the type does not have a size or is not a
+  /// primitive type.
+  ///
+  /// If this is a scalable vector type, the scalable property will be set and
+  /// the runtime size will be a positive integer multiple of the base size.
+  ///
+  /// Note that this may not reflect the size of memory allocated for an
+  /// instance of the type or the number of bytes that are written when an
+  /// instance of the type is stored to memory. The DataLayout class provides
+  /// additional query functions to provide this information.
+  ///
+  TypeSize getPrimitiveSizeInBits() const {
+    return LLVMTy->getPrimitiveSizeInBits();
+  }
+
+  /// If this is a vector type, return the getPrimitiveSizeInBits value for the
+  /// element type. Otherwise return the getPrimitiveSizeInBits value for this
+  /// type.
+  unsigned getScalarSizeInBits() const { return LLVMTy->getScalarSizeInBits(); }
+
+  /// Return the width of the mantissa of this type. This is only valid on
+  /// floating-point types. If the FP type does not have a stable mantissa (e.g.
+  /// ppc long double), this method returns -1.
+  int getFPMantissaWidth() const { return LLVMTy->getFPMantissaWidth(); }
+
+  /// Return whether the type is IEEE compatible, as defined by the eponymous
+  /// method in APFloat.
+  bool isIEEE() const { return LLVMTy->isIEEE(); }
+
+  /// If this is a vector type, return the element type, otherwise return
+  /// 'this'.
+  Type *getScalarType() const;
+
+  // TODO: ADD MISSING
+
+  static Type *getInt64Ty(Context &Ctx);
+  static Type *getInt32Ty(Context &Ctx);
+  static Type *getInt16Ty(Context &Ctx);
+  static Type *getInt8Ty(Context &Ctx);
+  static Type *getInt1Ty(Context &Ctx);
+  static Type *getDoubleTy(Context &Ctx);
+  static Type *getFloatTy(Context &Ctx);
+  // TODO: missing get*
+
+  /// Get the address space of this pointer or pointer vector type.
+  inline unsigned getPointerAddressSpace() const {
+    return LLVMTy->getPointerAddressSpace();
+  }
+
+#ifndef NDEBUG
+  void dumpOS(raw_ostream &OS) { LLVMTy->print(OS); }
+  LLVM_DUMP_METHOD void dump() {
+    dumpOS(dbgs());
+    dbgs() << "\n";
+  }
+#endif // NDEBUG
```

Comment:
What about print(raw_ostream &, bool Debug, bool Details)?

Reply:
Hmm we could wrap that too, but I don't think I have any use for it for now. In LLVM IR it seems to be used in a couple of places though.

Reply:
Sounds good.